src/doc/rustc/src/codegen-options/index.md

```diff
@@ -231,15 +231,22 @@ coverage measurement. Its use is not recommended.
 
 ## link-self-contained
 
-On `windows-gnu`, `linux-musl`, and `wasi` targets, this flag controls whether the
-linker will use libraries and objects shipped with Rust instead of those in the system.
-It takes one of the following values:
+This flag controls whether the linker will use libraries and objects shipped with Rust instead
+of those in the system. This allows overriding cases when detection fails or user wants to use shipped
+libraries.
+
+You can enable or disable the usage of any self-contained objects using one of the following values:
 
 * no value: rustc will use heuristic to disable self-contained mode if system has necessary tools.
 * `y`, `yes`, `on`, `true`: use only libraries/objects shipped with Rust.
 * `n`, `no`, `off` or `false`: rely on the user or the linker to provide non-Rust libraries/objects.
 
-This allows overriding cases when detection fails or user wants to use shipped libraries.
+It is also possible to enable or disable specific self-contained objects in a more granular way.
+You can pass a comma-separated list of self-contained objects, individually enabled (`+object`) or
+disabled (`-object`).
+
+Currently, only the `linker` granular option is stabilized:
+- `linker`: toggle the usage of self-contained linker objects (linker, dlltool, and their necessary libraries)
```

Comment:
```suggestion
- `linker`: toggle the usage of self-contained linker binaries (linker, dlltool, and their necessary libraries)
```
"Objects" above was about the shipped `.o` files specifically, maybe better use "binaries" or "components" in the paragraph above too.

Reply:
This was an oversight on my part, I had noticed the incorrect use of "object" in this new context, but the fix got lost (maybe during rebases). I've updated these paragraphs to make use of "components" when discussing the individual options, and "binaries" for the `linker` option.